.github/workflows/canary-integration-test.yml

```diff
@@ -57,20 +57,28 @@ jobs:
         kubectl -n rook-ceph exec $toolbox -- mkdir -p /etc/ceph/test-data
         kubectl -n rook-ceph cp tests/ceph-status-out $toolbox:/etc/ceph/test-data/
         kubectl -n rook-ceph cp deploy/examples/create-external-cluster-resources.py $toolbox:/etc/ceph
-        # remove the existing client auth that will be re-created by external script
-        kubectl -n rook-ceph exec $toolbox -- ceph auth del client.csi-cephfs-node
-        kubectl -n rook-ceph exec $toolbox -- ceph auth del client.csi-cephfs-provisioner
-        kubectl -n rook-ceph exec $toolbox -- ceph auth del client.csi-rbd-node
-        kubectl -n rook-ceph exec $toolbox -- ceph auth del client.csi-rbd-provisioner
+        timeout 10 sh -c "until kubectl -n rook-ceph exec $toolbox -- python3 /etc/ceph/create-external-cluster-resources.py --rbd-data-pool-name replicapool; do echo 'waiting for script to succeed' && sleep 1; done"
         # print existing client auth
         kubectl -n rook-ceph exec $toolbox -- ceph auth ls
-        timeout 10 sh -c "until kubectl -n rook-ceph exec $toolbox -- python3 /etc/ceph/create-external-cluster-resources.py --rbd-data-pool-name replicapool; do echo 'waiting for script to succeed' && sleep 1; done"
 
     - name: dry run external script create-external-cluster-resources.py
       run: |
         toolbox=$(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[*].metadata.name}')
         kubectl -n rook-ceph exec $toolbox -- python3 /etc/ceph/create-external-cluster-resources.py --rbd-data-pool-name=replicapool --dry-run
 
+    - name: test external script create-external-cluster-resources.py if users already exist with different caps
+      run: |
+        toolbox=$(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[*].metadata.name}')
+        # update client.csi-rbd-provisioner csi user caps
+        # print client.csi-rbd-provisioner user before update
+        kubectl -n rook-ceph exec $toolbox -- ceph auth get client.csi-rbd-provisioner
+        kubectl -n rook-ceph exec $toolbox -- ceph auth caps client.csi-rbd-provisioner mon 'profile rbd, allow command "osd blocklist"' osd 'profile rbd' mgr 'allow rw'
```

Comment:
are these caps different than the default caps from the script?

Reply:
Yes,
Updated ```mon 'profile rbd, allow command "osd blocklist"'``` with blocklist

Reply:
Check `test external script create-external-cluster-resources.py if users already exist with different caps` result:
```
[client.csi-rbd-provisioner]
	key = AQCRNP1hjk2TBxAAM4bpmHVbQoR/EiRikQoUhA==
	caps mgr = "allow rw"
	caps mon = "profile rbd"
	caps osd = "profile rbd"
```

and after
```
[client.csi-rbd-provisioner]
	key = AQCRNP1hjk2TBxAAM4bpmHVbQoR/EiRikQoUhA==
	caps mgr = "allow rw"
	caps mon = "profile rbd, allow command \"osd blocklist\""
	caps osd = "profile rbd"
```

Reply:
Why do you change the caps?

What we need to verify:

* create a `client.csi-rbd-provisioner` with different caps than the ones in the script, **before** the script run. For instance use `mon 'profile rbd, allow command "osd ls"'`
* run the script, it should not have any errors and report the current caps
* print the `ceph auth list` if we want

Reply:
> Why do you change the caps?

we need it because Rook already creates it before running the external-script.

Reply:
Done